[Lee Ming De] iP

LeeMingDe opened the pull request.

Commit:
Add Gradle support

LeeMingDe committed:
Added greet() method

LeeMingDe committed:
Added start() method

LeeMingDe committed:
Added blah() method

LeeMingDe committed:
Changed greet() to non-static

LeeMingDe committed:
Changed start() method to store text

LeeMingDe committed:
Changed lst() to display text

LeeMingDe committed:
Added Task as new class

LeeMingDe committed:
Added functions for task to be marked as done

LeeMingDe committed:
Level -3

LeeMingDe committed:
Level-4

LeeMingDe committed:
Level-4

LeeMingDe committed:
A-TextUiTesting

LeeMingDe committed:
A-TextUiTesting

LeeMingDe committed:
Level-5

Added Duke class and Exceptions

LeeMingDe committed:
Level-6

Added support for delete

LeeMingDe committed:
Changed Constants to Enum

A-Enums

LeeMingDe committed:
Fixed bug with enum

LeeMingDe committed:
Create duke.txt

LeeMingDe committed:
Create duke.txt

LeeMingDe committed:
Add getPreviousTask method

LeeMingDe committed:
Add writeToFile method

LeeMingDe committed:
Move writeToFile and getPreviousTask to another class

LeeMingDe committed:
Add LocalDateTime object to store deadlines

LeeMingDe committed:
Update Deadlines.java

LeeMingDe committed:
Merge branch 'branch-Level-7'

LeeMingDe committed:
Merge branch 'branch-Level-8'

LeeMingDe committed:
Remove Getters and Setters

LeeMingDe committed:
Change names of classes

LeeMingDe committed:
Fix bug in WriteData and change name

LeeMingDe committed:
Handle case for folder not existing

LeeMingDe committed:
Complete Storage class

LeeMingDe committed:
Complete Parser class

LeeMingDe committed:
Extract classes from an abstract Command class

LeeMingDe committed:
Change DateTimeFormatter for Deadline class

LeeMingDe committed:
Add UndoCommand class

LeeMingDe committed:
Divide classes into packages

LeeMingDe committed:
Extend Exception class to more subclasses

LeeMingDe committed:
Add JUnit test case for commands

LeeMingDe committed:
Add JUnit test case for tasks

LeeMingDe committed:
A-Jar

LeeMingDe committed:
Change method names in JUnit test

Change to follow java naming conventions for JUnit

LeeMingDe committed:
Add java docs to command package

LeeMingDe committed:
Add java docs to classes in task package

LeeMingDe committed:
Add java docs to classes

Java docs added to classes in exception package, Duke.java, Parser.java, Storage.java, Ui.java and TaskList.java

LeeMingDe committed:
A-CodingStandard

LeeMingDe committed:
Add Find feature to chatbot

Created new class called FindCommand and updated the methods in Parser.java and Ui.java

LeeMingDe committed:
Merge branch 'branch-A-CodingStandard'

LeeMingDe committed:
Merge branch 'Level-9'

LeeMingDe committed:
Add java docs to methods and classes

LeeMingDe committed:
Update Parser.java

LeeMingDe committed:
Shift all print method into Ui class

Adopt more oop practices by shifting all the print method into Ui class for display

LeeMingDe committed:
Merge branch 'add-gradle-support'

LeeMingDe committed:
Change package name for all classes

Package name was wrong after adding gradle support

LeeMingDe committed:
Rearrange code according to checkstyle

LeeMingDe committed:
Create simple GUI using JavaFX

Added code for JavaFX to run a GUI that displays hello world

LeeMingDe committed:
Add feature to echo the user

LeeMingDe committed:
Add custom behavior to dialog box

Added method to flip the dialog box so that Duke and user's output can be differentiated

LeeMingDe committed:
Rebuild scene using FXML

LeeMingDe committed:
Change all print messages to return a string

LeeMingDe committed:
Update GUI to display greetings

jacetjy commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,158 @@
+     * @return String containing the task added.
+     */
+    public String showTask(Task task, int numTask) {
+        String output = new StringBuilder().append(" Got it. I've added this task: \n")
```
Maybe you could separate this (eg. by assigning String.format to a variable first), so the line wouldn't be too long.

LeeMingDe replied:
oh okay sure!

jacetjy reviewed (commented):
Very well written code, I really couldn't find anything that violated coding standards, and the code quality was great. Classes and methods are well abstracted, names of variables/methods are easily understood. Occasionally a slightly complicated expression is used to parse strings, but does not affect understanding of what the code does. Javadocs is also well-written and helps with understanding of some methods. Overall, really pleasant to read, nice job :)

LeeMingDe committed:
Fix bug in MainWindow

Changed the path from /image/DaUser to /image/daUser

LeeMingDe committed:
Add Assertions in Storage

LeeMingDe committed:
Do some code refactoring

Refactor code in Parser to ensure that main path is not nested

Abstract out methods in Storage to improve readability

LeeMingDe committed:
Improve code quality

Abstracted out methods for MainWindow.java

Moved main path out and ensure it is not nested for classes in the Command Package.

LeeMingDe committed:
Add lambdas functions to iterate through lists

LeeMingDe committed:
Merge pull request #2 from LeeMingDe/branch-A-CodeQuality

Improve code quality

LeeMingDe committed:
Merge branch 'master' into branch-A-Lambdas

LeeMingDe committed:
Merge pull request #3 from LeeMingDe/branch-A-Lambdas

Add lambdas functions to iterate through lists

LeeMingDe committed:
Merge branch 'master' into branch-A-Assertions

LeeMingDe committed:
Merge pull request #1 from LeeMingDe/branch-A-Assertions

Add Assertions in Storage

LeeMingDe committed:
Create gradle.yml

LeeMingDe committed:
Fix bug in FindCommand

LeeMingDe committed:
Fix bug in FindCommand

fyshhh commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,39 @@
+ */
+public class Event extends Task {
+
+    protected String at;
```
Similarly, I think you can consider using dateAt instead of at so it's clearer what the variable is referring to.

fyshhh commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,57 @@
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.date = by;
```
I think you can consider using dateBy instead of by so it's clearer what the variable is referring to.

fyshhh reviewed (commented):
Overall, good work!

LeeMingDe committed:
Create RecurringTask class

LeeMingDe committed:
Finish implementing Recurring Task function

LeeMingDe committed:
Add java docs

LeeMingDe committed:
Merge branch 'branch-B-RecurringTask'

LeeMingDe committed:
Remove unused imports

LeeMingDe committed:
Improve GUI for app

Added shadows to dialogbox
Re-centered text

LeeMingDe committed:
Add Ui image into docs folder

LeeMingDe committed:
Update Readme

LeeMingDe committed:
Fix some minor bugs

LeeMingDe committed:
Merge branch 'master' of https://github.com/LeeMingDe/ip

LeeMingDe committed:
Update Readme

LeeMingDe committed:
Update Readme

LeeMingDe committed:
Set theme jekyll-theme-slate

LeeMingDe committed:
Update Readme

LeeMingDe committed:
Update Readme

LeeMingDe committed:
Update Readme

LeeMingDe committed:
Update README.md

LeeMingDe committed:
Fix bugs

LeeMingDe committed:
Change to System exit

LeeMingDe committed:
Update README.md
